app/scripts/views/settings/clients.js

```diff
@@ -130,16 +130,17 @@ define(function (require, exports, module) {
 
     _onDisconnectClient: function (event) {
       var item = this._attachedClients.get($(event.currentTarget).data('id'));
-
-      return this.user.destroyAccountClient(this.user.getSignedInAccount(), item)
-        .then(() => {
-          var clientType = item.get('clientType');
-
-          this.logViewEvent(clientType + '.disconnect');
-          if (clientType === Constants.CLIENT_TYPE_DEVICE && item.get('isCurrentDevice')) {
-            this.navigateToSignIn();
-          }
+      var clientType = item.get('clientType');
+      this.logViewEvent(clientType + '.disconnect');
+      // if a device then ask for confirmation
+      if (clientType === Constants.CLIENT_TYPE_DEVICE) {
+        this.navigate('settings/clients/disconnect', {
+          clients: this._attachedClients,
```

Comment:
This seems to be where it'd be simpler to just pass in the client since neither `clients` nor the `id` are used for anything except fetching the client.